modules/mesos/src/main/java/org/apache/ignite/mesos/resource/IgniteProvider.java

```diff
@@ -52,40 +75,93 @@ public IgniteProvider(String downloadFolder) {
 
     /**
      * @param ver Ignite version.
-     * @throws IOException If downloading failed.
      * @return Path to latest ignite.
+     * @throws IOException If downloading failed.
      */
     public String getIgnite(String ver) throws IOException {
         return downloadIgnite(ver);
     }
 
     /**
      * @param ver Ignite version which will be downloaded. If {@code null} will download the latest ignite version.
-     * @throws IOException If downloading failed.
      * @return Ignite archive.
+     * @throws IOException If downloading failed.
      */
     public String downloadIgnite(String ver) throws IOException {
         assert ver != null;
 
         URL url;
 
+        // get the latest version.
         if (ver.equals(ClusterProperties.DEFAULT_IGNITE_VERSION)) {
-            URL updateUrl = new URL(DOWNLOAD_LINK);
+            ver = findLatestVersion();
```

Comment:
We really need try {} around all this logic, if catch() then log error then use DOWNLOAD_URL_PATTERN.

Reply:
@alamar Agree, good point.

Reply:
How does it work currently?

Reply:
@alamar Currently (before the fix), 
1. if a user specifies the latest version (which is default) it tries to download the latest stable version from http://ignite.run/, which was discarded several versions ago (and therefore fails.)
2. Otherwise it uses the version specified by the user from the archives.
Please note, that a user can specify a URL for his/her custom builds with IGNITE_PACKAGE_URL.

Basically, I preserve this behavior, just fixed the download path.